feat(QDOG-54): Create login page

arasantos opened the pull request:
- Add login page to the app.tsx so it shows in path /
- Add design tokens and keep global styles in globals.scss
- Add fonts in layout.tsx
- Add responsive style for login page in page.module.scss, mostly matching figma
- Add images for pet owner and vet, can be temporary.

Mobile
<img width="441" height="773" alt="image" src="https://github.com/user-attachments/assets/5b4651f4-457a-4062-b7ef-7f50ba013432" />


Desktop
Owner:
<img width="1919" height="857" alt="image" src="https://github.com/user-attachments/assets/47377611-e4d3-4b4b-a9e8-fb70bb797c6f" />

Vet:
<img width="1905" height="859" alt="image" src="https://github.com/user-attachments/assets/3c15f29d-d9a2-4a1c-9ca0-e4eba4f83043" />

Validations:
<img width="1918" height="857" alt="image" src="https://github.com/user-attachments/assets/4102fade-c9a3-4caa-95b8-9dc10611a45a" />

<img width="1916" height="851" alt="image" src="https://github.com/user-attachments/assets/a072aac0-3eb7-4a59-a56a-9261509fa5ae" />

arasantos committed:
feat(QDOG-54): Create login page

Add login page to the app.tsx so it shows in path /
Add design tokens and keep global styles in globals.scss
Add fonts in layout.tsx
Add responsive style for login page in page.module.scss, mostly matching figma
Add images for pet owner and vet, can be temporary.

arasantos committed:
feat(QDOG-54): Create login page

Remove unused styles in globals.scss

JordonH03 commented on src/app/layout.tsx:
```diff
@@ -1,34 +1,28 @@
 
-import type { Metadata } from "next";
-import { Geist, Geist_Mono } from "next/font/google";
+import { Roboto, Tsukimi_Rounded } from "next/font/google";
```
Could we import DM Sans instead of Roboto for the body text? So it's aligned with our designs?

arasantos replied:
Thanks for letting me know! I  changed it and updated the images in pr description to reflect the changes

JordonH03 commented on src/app/page.tsx:
```diff
@@ -1,95 +1,70 @@
+  let emailErrorMessage: string = "Incorrect email format.";
+  let passwordErrorMessage: string = "Incorrect password.";
+
+  let emailPlaceholder: string = "youremail@email.com";
```
Mantine has a couple form hooks in `@mantine/form` that might be useful for client-side validations

arasantos replied:
I wonder if we should do validations in front end - I thought we'd be doing that in back end and then just showing the exception message or something in front end. I see the benefit in doing it in front end as well though

JordonH03 replied:
I was also expecting validation in the backend, but I thought having frontend validation would be nice on top so that the user could get feedback before hitting the api. We can discuss that more in meeting

arasantos replied:
Just so other reviewers are aware, we discussed this offline and concluded we should have validations for both front end and back end.

Anyway, I updated this PR to include validations. Updated the images of PR to show changes as well. :)

arasantos committed:
feat(QDOG-54): Create login page

Change Roboto font to DM Sans to accurately match Figma font
Add more type safety for variables and constants

arasantos committed:
feat(QDOG-54): Create login page

Add validations for email and password
Use form from Mantine in login form
Change config to allow allowImportingTsExtensions so we can import custom functions that we made

arasantos committed:
feat(QDOG-54): Create login page

Add type safety and more correct var names for validation.ts
